ohttp-server/src/main.rs

```diff
@@ -630,52 +653,392 @@ mod general_purpose_tests {
+            .await
+            .expect("Could not create new ohttp client builder");
+
+        let mut url: String = "http://localhost:9443/scoreee".to_string();
+        let mut target_path: String = TARGET_PATH.to_string();
+        let headers: Vec<String> = vec![];
+        let mut form_fields: Vec<String> = vec![String::from("file=@../examples/audio.mp3")];
+        let outer_headers: Vec<String> = vec![];
+
         let mut response = ohttp_client
             .post(&url, &target_path, &headers, &form_fields, &outer_headers)
             .await
             .unwrap();
+        let status = response.status();
+        info!("status: {status}");
+        assert!(!status.is_success());
+
+        let ohttp_client = OhttpClientBuilder::new()
+            .config(&hex_arg)
+            .build()
+            .await
+            .expect("Could not create new ohttp client builder");
+
+        url = URL_SCORE.to_string();
+        target_path = "/whisperrr".to_string();
+
+        response = ohttp_client
+            .post(&url, &target_path, &headers, &form_fields, &outer_headers)
+            .await
+            .expect("Could not post to scoring endpoint");
+
+        let status = response.status();
+        info!("status: {status}");
+        assert!(!status.is_success());
+
+        let ohttp_client = OhttpClientBuilder::new()
+            .config(&hex_arg)
+            .build()
+            .await
+            .expect("Could not create new ohttp client builder");
+
+        target_path = TARGET_PATH.to_string();
+        form_fields = vec![String::from("file=@../examples/audioo.mp3")];
+
+        if ohttp_client
+            .post(&url, &target_path, &headers, &form_fields, &outer_headers)
+            .await
+            .is_ok()
+        {
+            panic!("This should never happen!");
+        }
+
+        shutdown_server(server_handle, shutdown_channel_sender).await;
+    }
+
+    const DEFAULT_KMS_URL_CLIENT: &str =
+        "https://accconfinferenceprod.confidential-ledger.azure.com";
+    const DEFAULT_KMS_URL_SERVER: &str =
+        "https://accconfinferenceprod.confidential-ledger.azure.com/app/key";
+    const DEFAULT_MAA_URL: &str = "https://maanosecureboottestyfu.eus.attest.azure.net";
+
+    use std::{fs::File, io::Write, path::PathBuf};
+
+    async fn get_kms_cert() -> Option<PathBuf> {
+        let client = reqwest::Client::builder()
+            .danger_accept_invalid_certs(true) // Equivalent to -k in curl
+            .build()
+            .expect("reqwest::Client::builder() failed");
+
+        let response = client
+            .get("https://accconfinferenceprod.confidential-ledger.azure.com/node/network")
+            .send()
+            .await
+            .expect("reqwest::Client::get() failed");
+
+        let body: Value = response.json().await.expect("Failed to read response");
+        info!("body = {:?}", body);
+
+        // Accessing the "name" field
+        if let Value::Map(map) = body {
```

Comment:
Wonder if its possible to write this big nested if better, perhaps for a later PR?